### PR TITLE
Add 'get_battery_info' method to the SoCo class

### PR DESCRIPTION
**Pull request by pwt**
This PR addresses #723.

I've only tested against a Move on my S2 system, and there are indications that this approach doesn't work with S1 systems. It would be useful if someone could test this.

### CODE REVIEW

**Comment by KennethNielsen** [comment]
I think I have a mixed S1 and S2, but I don't have any devices with battery :(In any case.

**Comment by pwt** [comment]
> I think I have a mixed S1 and S2, but I don't have any devices with battery :(

I factory reset my Move and attempted to add it to my S1 system for testing. However, it looks like there's a bug in the current Sonos S1 iOS/iPadOS controller apps, meaning the process gets stuck when adding WiFi creds. Fortunately, I was able to add it back to my S2 system without any issues.

> In any case. Since the code reaches fairly wide, when the feature is not supported, I would be ok to merge it even without testing on S1 and then improve if anyone opens up an issue saying it doesn't work on S1 but should. There isn't much more we can do without more data or testing. What do you think @pwt ?

Yes, I agree. It's not as if it's a mission-critical feature :)

**Comment by KennethNielsen** [comment]
Do you want to add a software test for it or should I review as is?

**Comment by pwt** [comment]
> Do you want to add a software test for it or should I review as is?

I'm not planning to add a test for this one.

**Comment by pwt** [comment]
Note that I could test explicitly for whether the target speaker is a Move, and avoid the network call if it isn't.

I chose not to do this because if additional battery-powered speakers arrive, there's a good chance that this call will work without change for these new speakers.

File: soco/core.py [hunk 1]
```diff
@@ -2153,6 +2158,70 @@ def get_sonos_playlist_by_attr(self, attr_name, match):
...
+            timeout (float, optional): The timeout to use when making the
+                HTTP request.
+
+        Returns:
+            dict: A `dict` containing battery status data.
+
+            Example return value::
+
+                {'Health': 'GREEN',
+                 'Level': '100',
```
Comment (KennethNielsen) [inline soco/core.py:2183]: Should we maybe parse the Level and make it an int?

**Review by KennethNielsen — APPROVED** [review]
I have one minor suggestion, but it is approved as is if you don't agree.

File: soco/core.py (hunk 1 above)
Comment (pwt) [reply soco/core.py:2183]: I did consider reaching into the battery data, but concluded that it's less brittle just to gather what's there and pass it back without making any assumptions.

Comment (pwt) [reply soco/core.py]: OK, if "Level" is present in the dict, it now gets converted to an int. Will still work if the fields change.

**Pull request merged**